cloudflare-tag-purge.php

```diff
@@ -9,251 +9,271 @@
-    $post = get_post($post_id);
-
-    if ($post instanceof WP_Post) {
-        // add author page
-        if (isset($post->post_author) && !empty($post->post_author)) {
-            $tags = array_merge([get_yoast_cache_prefix() . 'author-' . $post->post_author], $tags);
-        }
-
-        $post_type = get_post_type($post->ID);
-        $taxonomies = get_object_taxonomies($post_type);
-        foreach ($taxonomies as $tax) {
-            foreach (wp_get_object_terms($post->ID, $tax) as $taxonomy_details) {
-                $tags[] = get_yoast_cache_prefix() . $tax . '-' . $taxonomy_details->term_id;
-                $tags[] = get_yoast_cache_prefix() . $tax . '-' . $taxonomy_details->slug;
-            }
-        }
-    }
-
-    do_action_ref_array('yoast_cloudflare_purge_cache_tags', $tags);
-
-    return $tags;
+define( 'cloudflare_tag_purge_version', '1.0.5' );
+define( 'YOAST_CLOUDFLARE_LOG_STATUS_INFO', 'info' );
+define( 'YOAST_CLOUDFLARE_LOG_STATUS_PURGE', 'purge' );
+
+class Yoast_CloudFlare_Purge {
+	/**
+	 * @var string
+	 */
+	private $auth_key;
+
+	/**
+	 * @var string
+	 */
+	private $mail;
+
+	/**
+	 * @var string
+	 */
+	private $zone_id;
+
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+		$this->auth_key = $this->read_environment_setting( 'CF_KEY', null );
+		$this->mail     = $this->read_environment_setting( 'CF_EMAIL', null );
+		$this->zone_id  = $this->read_environment_setting( 'CF_ZONE_ID', null );
+
+		if ( empty( $this->auth_key ) ) {
+			$this->cache_tag_log( YOAST_CLOUDFLARE_LOG_STATUS_INFO, [
+				'purge' => false,
+				'info'  => 'No CloudFlare key available.'
+			] );
+
+			return;
+		}
+
+		if ( empty( $this->mail ) || empty( $this->zone_id ) ) {
+			$this->cache_tag_log( YOAST_CLOUDFLARE_LOG_STATUS_INFO, [
+				'purge' => false,
+				'info'  => 'No CloudFlare email or zone ID available.'
+			] );
+
+			return;
+		}
+
+		add_action( 'plugin_loaded', [ $this, 'register_hooks' ] );
+	}
+
+	/**
+	 * Register hooks.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		if ( ! is_admin() ) {
+			add_filter( 'body_class', [ $this, 'cache_prefix_body_class' ], 40, 1 );
+		}
+
+		if ( is_admin() || wp_is_json_request() ) {
+			add_action( 'edit_post', [ $this, 'purge_action_edit_post' ], 10, 2 );
+		}
+	}
+
+	/**
+	 * @param int     $post_id
+	 * @param WP_Post $post
+	 *
+	 * @return void
+	 */
+	public function purge_action_edit_post( int $post_id, WP_Post $post ) {
+		$this->execute_tag_purge( (int) $post_id, $post );
+	}
+
+
+	/**
+	 * @param $key
+	 * @param $default
+	 *
+	 * @return mixed
+	 */
+	private function read_environment_setting( $key, $default ) {
+		if ( ! empty( getenv( $key ) ) ) {
+			return getenv( $key );
+		}
+
+		return $default;
+	}
+
+	/**
+	 * @param int $post_id
+	 *
+	 * @return array|string[]
+	 */
+	function get_cache_tags_by_post_id( int $post_id ) {
+		$tags = [ $this->get_cache_prefix() . 'postid-' . (int) $post_id ];
```

Comment:
No need to cast with the typed argument
```suggestion
		$tags = [ $this->get_cache_prefix() . 'postid-' . $post_id ];
```